src/components/Form.test.js

```diff
@@ -305,11 +330,209 @@ describe("Form", () => {
+      expect(screen.queryByDisplayValue("22")).not.toBeInTheDocument();
+
+      userEvent.type(screen.getByLabelText("Name"), "Helena");
+      userEvent.type(screen.getByLabelText("Street number"), "22");
+
+      await waitFor(() => {
+        expect(
+          screen.queryByText("Please enter a name")
+        ).not.toBeInTheDocument();
+        expect(
+          screen.queryByText("Please enter a street number")
+        ).not.toBeInTheDocument();
+        expect(screen.queryByDisplayValue("Helena")).toBeInTheDocument();
+        expect(screen.queryByDisplayValue("22")).toBeInTheDocument();
+      });
+
+      userEvent.click(screen.getByTestId("resetButton"));
+
+      await waitFor(() => {
+        expect(screen.queryByText("Please enter a name")).toBeInTheDocument();
+        expect(
+          screen.queryByText("Please enter a street number")
+        ).toBeInTheDocument();
+        expect(screen.queryByDisplayValue("Helena")).not.toBeInTheDocument();
+        expect(screen.queryByDisplayValue("22")).not.toBeInTheDocument();
+      });
+    });
+
+    it("sets form state to new values when resetForm is called with argument", async () => {
+      const initialValues = {
+        name: "Helena",
+        address: { streetNumber: "22", streetName: "" },
+      };
+      const initialErrors = {};
+
+      const newValues = {
+        name: "Bob",
+        address: { streetNumber: "1", streetName: "" },
+      };
+      const newErrors = {
+        name: ["Please enter a name"],
+        address: { streetNumber: ["Please enter a street number"] },
+      };
+
+      const RenderChild = ({ resetForm }) => (
+        <>
+          <Input name="name" label="Name" />
+          <Input name="address.streetNumber" label="Street number" />
+          <Input name="address.streetName" label="Street name" />
+          <Button
+            testId="resetButton"
+            onClick={() => {
+              resetForm({ values: newValues, errors: newErrors });
+            }}
+          >
+            Reset
+          </Button>
+        </>
+      );
+
+      render(
+        <Form initialValues={initialValues} initialErrors={initialErrors}>
+          {RenderChild}
+        </Form>
+      );
+
+      expect(screen.queryByText("Please enter a name")).not.toBeInTheDocument();
+      expect(
+        screen.queryByText("Please enter a street number")
+      ).not.toBeInTheDocument();
+      expect(screen.queryByDisplayValue("Helena")).toBeInTheDocument();
+      expect(screen.queryByDisplayValue("22")).toBeInTheDocument();
+
+      userEvent.click(screen.getByTestId("resetButton"));
+
+      await waitFor(() => {
+        expect(screen.queryByText("Please enter a name")).toBeInTheDocument();
+        expect(
+          screen.queryByText("Please enter a street number")
+        ).toBeInTheDocument();
+        expect(screen.queryByDisplayValue("Bob")).toBeInTheDocument();
+        expect(screen.queryByDisplayValue("1")).toBeInTheDocument();
```

Comment:
This would be more specific:
```
expect(screen.getByLabelText("Name")).toHaveValue("Bob");
expect(screen.getByLabelText("Street number")).toHaveValue("1");
```